inst/perl/lib/BiodiverseR.pm

```diff
@@ -268,6 +325,27 @@ $log->debug("Called startup");
         );
     }
 
+    # Check if the api_key sent with the call is the same as api_key stored. If not return a string containing the error message.
+    sub check_api_key ($c, $api_key) {
+        my $body_params = $c->req->json;
+        my $sent_api_key = $body_params->{api_key};
+        my $perl_stored_api_key = $api_key;
+        # $log->debug("Stored api key is:");
+        # $log->debug($perl_stored_api_key);
+        # $log->debug("Sent api key is:");
+        # $log->debug($sent_api_key);
+        
+        # Keys are undefined in Perl tests but I believe thats because its not calling anything from the R server so it makes sense there are no keys
+        $sent_api_key //= "";
+
+        $perl_stored_api_key //= "";
+
+        return "Stored api_key does not match api_key passed in"
```

Comment:
Perhaps this should raise an exception on failure.  Then any callers would need to wrap the calls in ```eval{}``` and check for errors.  (We could also use [Feature::Compat::Try](https://metacpan.org/pod/Feature::Compat::Try) but that's for later).

Reply:
I believe I have done what you asked. Every single call to check the api key is done through eval now.